src/instances/RaidInstance.ts

```diff
@@ -1517,6 +1527,234 @@ export class RaidInstance {
+        // Collect inputs
+
+        // Sorry but I am NOT using the interactioncollector we have
+        const collector = new InteractionCollector(Bot.BotInstance.client, {
+            channel: i.channelId,
+            interactionType: InteractionTypes.MODAL_SUBMIT,
+            time: 30_000,
+            max: 1,
+            filter: (modalInteraction: ModalSubmitInteraction) => {
+                const correctUser = i.user.id === modalInteraction.user.id;
+                if (!correctUser) modalInteraction.reply({ content: "You are not the leader of this raid.", ephemeral: true });
+                // should other raid leaders be able to chain log?
+                return correctUser;
+            }
+        });
+
+        collector.on("collect", async (modalInteraction: ModalSubmitInteraction) => {
+            // Log chain pops for the key user here.
+            const keyPopper = modalInteraction.fields.getTextInputValue("chain_log_name");
+            let amountPopped = Number.parseInt(modalInteraction.fields.getTextInputValue("chain_log_amount"));
+            if (Number.isNaN(amountPopped)) {
+                amountPopped = 1;
+            }
+
+            if (!trackedUser) {
+                const findResult = await UserManager.resolveMember(modalInteraction.guild!, keyPopper);
+                if (!findResult || !findResult?.member) {
+                    modalInteraction.reply({ content: "Couldn't find that user in this server.", ephemeral: true });
+                    return;
+                }
+                trackedUser = findResult.member;
+            }
+
+            if (!trackedDungeonKey) {
+                if (this._dungeon.keyReactions.length > 1) {
+                    modalInteraction.reply({ content: "This dungeon is too complex for now.", ephemeral: true });
+                    return;
+                }
+
+                trackedDungeonKey = this._dungeon.keyReactions[0].mapKey;
+            }
+
+            // this should probably be logged in a channel somewhere for admins to see
+            LOGGER.info(`[${modalInteraction.guild?.name}] ${(modalInteraction.member as GuildMember).displayName}`
+                + ` just logged ${amountPopped} ${trackedDungeonKey}s for ${trackedUser.displayName}`);
+            LoggerManager.logKeyUse(trackedUser!, trackedDungeonKey!, amountPopped);
+
+            // Try to get completes for members
+            const membersAtEnd: GuildMember[] = [];
+            const membersThatLeft: GuildMember[] = [];
+
+            const BUTTONS = new MessageActionRow();
+            if (!this._vcless) {
+                const VC_USERS_BUTTON = new MessageButton()
+                    .setCustomId("chain_log_use_vc")
+                    .setLabel("Users in VC")
+                    .setEmoji("🎙️")
+                    .setStyle("PRIMARY");
+
+                BUTTONS.addComponents(VC_USERS_BUTTON);
+            }
+            const SKIP_BUTTON = new MessageButton()
```

Comment:
Use camelcase here, or define a constant and import the constant. For what it's worth, `AdvancedCollector` has a helper function `cloneButton` that you might find helpful. Use the function to clone the button and then, if needed, assign a new custom ID.